sigopt/hyperopt/base.py

```diff
@@ -0,0 +1,108 @@
+from .compat import Trials, STATUS_OK, STATUS_FAIL
+from .. import SigOptFactory
+
+HYPEROPT_SOURCE_NAME = 'Hyperopt Suggest'
+HYPEROPT_SOURCE_PRIORITY = 1
+
+
+class SigOptTrials(object):
+  def __init__(self, project, trials=None, online=True):
+    self.factory = SigOptFactory(project)
```

Comment:
can we make sure that `project` here has the correct type? 
In particular, I'm thinking more about a nice error message if they try to do: `trials = SigOptTrials(trials)`

Reply:
`SigOptFactory(project)` will check `project` is valid or not.

Reply:
oh, I know that, but my point is that a typical Hyperopt user could be confused about the current error; it would be nice to have a helpful error message that says we expect a `project` name for organizing your Trials.